docs/features/software-catalog/catalog-customization.md

```diff
@@ -502,3 +502,201 @@ const routes = (
+
+```yaml title="Example usage of $all"
+filter:
+  $all:
+    - kind: component
+    - $not:
+      spec.type: service
+```
+
+#### `$any`
+
+The `$any` operator has the following syntax:
+
+```json
+{ $any: [ { <expression1> }, { <expression2> }, ...] }
+```
+
+The `$any` operator evaluates to `true` if at least one of the expressions within the provided array evaluate to `true`. This includes an empty array, which means that `{ "$any": [] }` always evaluates to `false`.
+
+```yaml title="Example usage of $any"
+filter:
+  $any:
+    - kind: component
+    - metadata.annotations.github.com/project-slug: { $exists: true }
+```
+
+#### `$not`
+
+The `$not` operator has the following syntax:
+
+```json
+{ $not: { <expression> } }
+```
+
+The `$not` operator inverts the result of the provided express. If the expression evalutes to `true` then `$not` will evaluate to false, and the other way around.
+
+```yaml title="Example usage of $not"
+filter:
+  $not:
+    kind: template
+```
+
+### Entity predicate value operators
+
+The following section lists all value operators for entity predicates.
+
+#### `$exists`
+
+The `$exists` operator has the following syntax:
+
+```json
+{ field: { $exists: <boolean> } }
+```
+
+The `$exists` operator will evaluate to `true` if the existence of the value it matches against matches the provided boolean. That is `{ $exists: true }` will evaluate to `true` if and only if the value is defined, and `{ $exists: false }` will evalute to `true` if and only if the value is `undefined`.
+
+```yaml title="Example usage of $exists"
+filter:
+  metadata.annotations.github.com/project-slug: { $exists: true }
+```
+
+#### `$in`
+
+The `$in` operator has the following syntax:
+
+```json
+{ field: { $in: [ <primitive1>, <primitive2>, ... ] } }
+```
+
+The `$in` operator will evaluate to `true` if the value it is matched against is exists within the array of primitives. The comparison is case insensitive and can only be done across primitive values. If the value matched against is an object or array, the operator will always evaluate to `false`.
+
+```yaml title="Example usage of $in"
+filter:
+  kind:
+    $in: [component, api]
+```
+
+#### `$contains`
+
+The `$contains` operator has the following syntax:
+
+```json
+{ field: { $contains: { <expression> } } }
+```
+
+The `$contains` operator will evaluate to `true` if the value it is matched against is an array, and at least one of the elements in the array fully matches the provided expression. If the value matched against is not an array, or if the array is empty, the operator will always evaluate to `false`.
+
+The expression used to match against the array can be any valid entity predicate expression, including logical operators and value operators.
+
+```yaml title="Example usage of $contains"
+filter:
+  relations:
+    $contains:
```

Comment:
Hm how would we translate this to sql? I guess this is at the point where we actually could not support it in a general sense right?

Reply:
Yep we'd need to update the storage